### PR TITLE
OCM-4021: Provider Attributes changes

### PR DESCRIPTION
**Pull request by nirarg**
* Remove unused attributes
* Remove all attributes but Token from the docs - internal attributes
* Add option for environment variables for all string attributes
* Add "Authentication and configuration" section in the main index and remove attributes section

### CODE REVIEW

**Review by EricPonvelle — COMMENTED** [review]
/lgtm

File: provider/provider.go [hunk 1]
```diff
@@ -130,6 +119,16 @@ func (p *Provider) Schema(ctx context.Context, req tfprovider.SchemaRequest, res
 	}
 }
 
+func (p *Provider) getValue(attr types.String, envSuffix string) (string, bool) {
```
Comment (sagidayan) [inline provider/provider.go:122]: Maybe change the name? it's a bit ambiguous.

File: provider/provider.go [hunk 2]
```diff
@@ -81,30 +79,21 @@ func (p *Provider) Schema(ctx context.Context, req tfprovider.SchemaRequest, res
 	resp.Schema = tfpschema.Schema{
 		Attributes: map[string]tfpschema.Attribute{
 			"url": tfpschema.StringAttribute{
-				Description: "URL of the API server.",
+				Description: fmt.Sprintf("URL sets the base URL of the API gateway. The default is `%s`", sdk.DefaultURL),
 				Optional:    true,
```
Comment (bardielle) [inline provider/provider.go:83]: maybe you can set here a default value?

Comment (nirarg) [reply provider/provider.go:83]: The default value is determined in the SDK level, no need to add it here as well

File: provider/provider.go (hunk 1 above)
Comment (nirarg) [reply provider/provider.go]: As discussed, I changed the function name to `getAttrValueOrConfig`

**Pull request merged**